core/services/pipeline/runner.go

```diff
@@ -75,23 +77,17 @@ func (r *runner) Stop() {
 	if err != nil {
 		logger.Error(err)
 	}
-
-	if r.newRunsListener != nil {
-		err := r.newRunsListener.Stop()
-		if err != nil {
-			logger.Errorw(`Error stopping pipeline runner's "new runs" listener`, "error", err)
-		}
-	}
-	<-r.chDone
 }
 
 func (r *runner) runLoop() {
 	defer close(r.chDone)
+	defer r.destroy()
 
-	var err error
-	r.newRunsListener, err = r.orm.ListenForNewRuns()
+	newRunsSubscription, err := r.orm.ListenForNewRuns()
 	if err != nil {
-		logger.Errorw(`Pipeline runner failed to subscribe to "new run" events, falling back to polling`, "error", err)
+		logger.Error("Pipeline runner could not subscribe to new run events, falling back to polling")
```

Comment:
Same here, is it even worth trying to continue if we couldnt even open the subscription?

Reply:
Same response as above.  Still works, just a little bit slower, and having a running node can help with debugging